process/outportBlockConverter.go

```diff
@@ -40,6 +42,349 @@ func NewOutportBlockConverter(
+	headerType := outportBlock.BlockData.HeaderType
+
+	// check if the header type is supported by this function.
+	if headerType != string(core.ShardHeaderV1) && headerType != string(core.ShardHeaderV2) {
+		return nil, fmt.Errorf("cannot convert to shard outport block. header type: %s not supported", headerType)
+	}
+
+	shardOutportBlock := &hyperOutportBlocks.ShardOutportBlockV2{}
+	shardOutportBlock.ShardID = outportBlock.ShardID
+	err := o.handleBlockData(outportBlock.BlockData, shardOutportBlock)
+	if err != nil {
+		return nil, fmt.Errorf("failed to migrate block data: %w", err)
+	}
+	err = o.handleTransactionPool(outportBlock.TransactionPool, shardOutportBlock)
+	if err != nil {
+		return nil, fmt.Errorf("failed to migrate transacion pool: %w", err)
+	}
+	handleHeaderGasConsumption(outportBlock.HeaderGasConsumption, shardOutportBlock)
+	handleAlteredAccounts(outportBlock.AlteredAccounts, shardOutportBlock)
+
+	shardOutportBlock.NotarizedHeadersHashes = outportBlock.NotarizedHeadersHashes
+	shardOutportBlock.NumberOfShards = outportBlock.NumberOfShards
+	shardOutportBlock.SignersIndexes = outportBlock.SignersIndexes
+	shardOutportBlock.HighestFinalBlockNonce = outportBlock.HighestFinalBlockNonce
+	shardOutportBlock.HighestFinalBlockHash = outportBlock.HighestFinalBlockHash
+
+	return shardOutportBlock, nil
+}
+
+func (o *outportBlockConverter) handleBlockData(blockData *outport.BlockData, shardOutportBlock *hyperOutportBlocks.ShardOutportBlockV2) error {
+	if blockData == nil {
+		return nil
+	}
+
+	var miniBlocks []*hyperOutportBlocks.MiniBlock
+	if blockData.Body != nil {
+		miniBlocks = copyMiniBlocks(blockData.Body.MiniBlocks)
+	}
+	intraShardMiniBlocks := copyMiniBlocks(blockData.IntraShardMiniBlocks)
+
+	shardOutportBlock.BlockData = &hyperOutportBlocks.BlockData{
+		ShardID:    blockData.ShardID,
+		HeaderType: blockData.HeaderType,
+		HeaderHash: blockData.HeaderHash,
+		Body: &hyperOutportBlocks.Body{
+			MiniBlocks: miniBlocks,
+		},
+		IntraShardMiniBlocks: intraShardMiniBlocks,
+	}
+
+	var err error
+	switch blockData.HeaderType {
+	case string(core.ShardHeaderV1):
+		err = o.handleHeaderV1(blockData.HeaderBytes, shardOutportBlock.BlockData)
+		if err != nil {
+			return fmt.Errorf("failed to handle header v1: %w", err)
+		}
+
+	case string(core.ShardHeaderV2):
+		err = o.handleHeaderV2(blockData.HeaderBytes, shardOutportBlock.BlockData)
+		if err != nil {
+			return fmt.Errorf("failed to handle header v2: %w", err)
+		}
+
+	default:
+		return fmt.Errorf("unknown header type [%s]", blockData.HeaderType)
+	}
+
+	return nil
+}
+
+func (o *outportBlockConverter) handleHeaderV1(headerBytes []byte, shardBlockData *hyperOutportBlocks.BlockData) error {
+	blockHeader := block.Header{}
+	err := o.gogoProtoMarshaller.Unmarshal(&blockHeader, headerBytes)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal: %w", err)
+	}
+
+	shardBlockData.Header = &hyperOutportBlocks.Header{}
+	header := shardBlockData.Header
```

Comment:
we can construct header variable at first, and the set to `shardBlockData.Header`, no need to set empty header first and then reference the header with a new variable

Reply:
optional suggestion: set simple fields directly when creating new header variable `&hyperOutportBlocks.Header{}` and set slices later on

Reply:
same applies to `handleShardV2`

Reply:
done,